aten/src/ATen/native/mkldnn/Matmul.cpp

```diff
@@ -127,11 +127,21 @@ void mkldnn_matmul(
               (mat1.dim() == 2 && mat2.dim() == 1) || // aten::mv
               (mat1.dim() == 1 && mat2.dim() == 1),  // aten::dot
               "mkldnn_matmul:  unsupported dims for mat and mat2");
-  TORCH_CHECK(mat1.scalar_type() == at::kBFloat16 &&
-   mat2.scalar_type() == at::kBFloat16 &&
-   result.scalar_type() == at::kBFloat16, "mkldnn_matmul:  only enabled for bf16 path");
+
   TORCH_CHECK(mkldnn_bf16_device_check(),
-    "mkldnn_matmul: mkldnn_matmul bf16 path needs the cpu support avx512bw, avx512vl and avx512dq");
+    "mkldnn_matmul: mkldnn_matmul bf16 path needs the cpu support avx512bw, avx512vl and avx512dq, or AWS Graviton3");
+
+  if (mkldnn_bf16_device_check_arm()) {
```

Comment:
Can ARM specific code be guarded under define to avoid even compiling this code on x86?
```suggestion
#ifdef __ARM__
  if (mkldnn_bf16_device_check_arm()) {
#else
 if (false) {
#endif
```